### PR TITLE
Add channel icon fields

### PR DESCRIPTION
**Pull request by aamiaa**
This documents `icon_emoji` and `theme_color` fields of channel object

### CODE REVIEW

File: pages/resources/channel.mdx [hunk 1]
```diff
@@ -45,6 +45,8 @@ Represents a guild or private channel within Discord.
...
 | default_sort_order?                 | ?integer                                                           | The default [sort order](#sort-order-type) of a thread-only channel's threads (default `LATEST_ACTIVITY`)                                                                                    |
+| icon_emoji?                         | ?[icon emoji](#icon-emoji-object) object                           | The emoji to show next to the channel name in channels list                                                                                                                                  |
+| theme_color?                        | ?integer                                                           | The background color of the channel icon emoji encoded as an integer representation of hexadecimal color code                                                                                |
```
Comment (splatterxl) [inline pages/resources/channel.mdx:49]:
```suggestion
| theme_color?                        | ?integer                                                           | The background color of the channel icon emoji encoded as an integer hex color code                                                                                |
```

File: pages/resources/channel.mdx [hunk 2]
```diff
@@ -485,6 +487,19 @@ An object that specifies the emoji to use as the default way to react to a `GUIL
 
 ^1^ At most one of `emoji_id` and `emoji_name` may be set to a non-null value.
 
+### Icon Emoji Object
+
+An object that specifies the emoji to use as the icon displayed next to a channel's name.
```
Comment (splatterxl) [inline pages/resources/channel.mdx:492]:
```suggestion
An object that specifies the emoji to use as the icon displayed next to a channel's name. One of the values must be null.
```

File: pages/resources/channel.mdx [hunk 3]
```diff
@@ -485,6 +487,19 @@ An object that specifies the emoji to use as the default way to react to a `GUIL
...
+
+###### Icon Emoji Structure
+
+| Field          | Type       | Description                        |
+| -------------- | ---------- | ---------------------------------- |
+| emoji_id ^1^   | ?snowflake | The ID of a guild's custom emoji   |
+| emoji_name ^1^ | ?string    | The unicode character of the emoji |
+
+^1^ At most one of `emoji_id` and `emoji_name` may be set to a non-null value.
```
Comment (splatterxl) [inline pages/resources/channel.mdx:501]: No need to make this a footnote if it applies to all of the fields.

**Review by splatterxl — CHANGES_REQUESTED** [review]

File: pages/resources/channel.mdx (hunk 3 above)
Comment (aamiaa) [reply pages/resources/channel.mdx:501]: I copy/pasted it from the `Default Reaction Object` section which is already written that way. If we change this here do we change it in both places?

File: pages/resources/channel.mdx (hunk 1 above)
Comment (aamiaa) [reply pages/resources/channel.mdx:49]: This is the way it's written everywhere else in the docs. I think it should be kept for consistency.

File: pages/resources/channel.mdx (hunk 3 above)
Comment (dolfies) [reply pages/resources/channel.mdx:501]: Is this even true here? I remember this field having some weird behavior (e.g. I'm pretty sure the entire object can be null AND both fields in the object can be null and both cases result in no icon)

Comment (aamiaa) [reply pages/resources/channel.mdx:501]: The [official documentation](https://discord.com/developers/docs/resources/channel#default-reaction-object) says "Exactly one of `emoji_id` and `emoji_name` must be set", and that also seems to be the error returned by the api:

image

Comment (aamiaa) [reply pages/resources/channel.mdx:501]: Setting both to `null`, as well as the entire field to `null` works too

Comment (dolfies) [reply pages/resources/channel.mdx:501]: Now that I think about it, can this object be merged with the above one into something more generic that can be used for both cases?

Comment (aamiaa) [reply pages/resources/channel.mdx:501]: Probably, but then we have to come up with a name for it which fits both uses and isn't confusing at the same time (so ex. can't be "Emoji Object"), and I don't think I have any ideas xd

Comment (dolfies) [reply pages/resources/channel.mdx:501]: partial emoji object is probably fine

**Review by dolfies — CHANGES_REQUESTED** [review]
This doesn't document the fields on modify channel or create channel (if applicable) <:bleh:1047549509745524777>

**Review by dolfies — APPROVED** [review]
ty pookie 🥰

**Pull request merged**